staging/src/kubevirt.io/api/instancetype/v1alpha2/types.go

```diff
@@ -77,11 +78,16 @@ type VirtualMachineClusterInstancetypeList struct {
 type VirtualMachineInstancetypeSpec struct {
 
 	// Required CPU related attributes of the instancetype.
+	// +optional
 	CPU CPUInstancetype `json:"cpu"`
 
 	// Required Memory related attributes of the instancetype.
+	// +optional
 	Memory MemoryInstancetype `json:"memory"`
 
+	// Resources describes the Compute Resources required by this vmi
+	Resources ResourceRequirements `json:"resources,omitempty"`
```

Comment:
You can directly use `v1.ResourceRequirements` type from `kubevirt.io/api/core/v1`.  Then you don't need to create the `ResourceRequirements` structure below.